sw/device/silicon_creator/manuf/base/perso_tlv_data.h

```diff
@@ -0,0 +1,112 @@
+// Copyright lowRISC contributors (OpenTitan project).
+// Licensed under the Apache License, Version 2.0, see LICENSE for details.
+// SPDX-License-Identifier: Apache-2.0
+
+#ifndef OPENTITAN_SW_DEVICE_SILICON_CREATOR_MANUF_BASE_PERSO_TLV_DATA_H_
+#define OPENTITAN_SW_DEVICE_SILICON_CREATOR_MANUF_BASE_PERSO_TLV_DATA_H_
+
+#include <stdbool.h>
+#include <stddef.h>
+#include <stdint.h>
+
+#include "sw/device/lib/base/status.h"
+#include "sw/device/lib/testing/json/provisioning_data.h"
+
+/**
+ * Personalization data is sent between the device and the host during the
+ * device provision. Personalization data is laid out as a sequence of
+ * concatenated LTV objects. The length and type field of the objects are packed
+ * in a 16 bit integer:
+ *  d15                                           d0
+ *  +-------------+--------------------------------+
+ *  | 4 bit type  |   12 bits total  object size   |
+ *  +-------------+--------------------------------+
+ *
+ * The following object types have been defined so far:
+ */
+typedef enum perso_object_type {
+  kPersoObjectTypeX509Tbs = 0,
+  kPersoObjectTypeX509Cert = 1,
+  kPersoObjectTypeDevSeed = 2,
+} perso_object_type_t;
+
+typedef uint16_t object_header_t;
+typedef uint16_t cert_header_t;
+typedef uint16_t dev_seed_header_t;
+
+typedef enum obj_header_fields {
+  // Object size, total size, this header included.
+  kObjhSizeFieldShift = 0,
+  kObjhSizeFieldWidth = 12,
+  kObjhSizeFieldMask = (1 << kObjhSizeFieldWidth) - 1,
+
+  // Object type, one of perso_object_types_t.
+  kObjhTypeFieldShift = kObjhSizeFieldWidth,
+  kObjhTypeFieldWidth = sizeof(object_header_t) * 8 - kObjhSizeFieldWidth,
+  kObjhTypeFieldMask = (1 << kObjhTypeFieldWidth) - 1,
+} obj_header_fields_t;
+
+/**
+ * The x509 certificate is prepended by a 16 bits header followed by the ASCII
+ * characters of the certificate name, followed by the certificate body.
+ *
+ * The certificate header includes 4 bits for the certificate name length then
+ * the full size of the certificate object (header size + name length +
+ * certificate size).
+ *
+ *  d15                                           d0
+ *  +-------------+--------------------------------+
+ *  | 4 bit length|       12 bits total size       |
+ *  +-------------+--------------------------------+
+ */
+typedef enum cert_header_fields {
+  // Certificate size, total size, this header and name length included.
+  kCrthSizeFieldShift = 0,
+  kCrthSizeFieldWidth = 12,
+  kCrthSizeFieldMask = (1 << kCrthSizeFieldWidth) - 1,
+
+  // Length of the certificate name immediately following the header.
+  kCrthNameSizeFieldShift = kCrthSizeFieldWidth,
+  kCrthNameSizeFieldWidth = sizeof(cert_header_t) * 8 - kCrthSizeFieldWidth,
+  kCrthNameSizeFieldMask = (1 << kCrthNameSizeFieldWidth) - 1,
+} cert_header_fields_t;
+
+// Helper macros allowing set or get various header fields.
+#define set_field(type_name, field_name, full_value, field_value)             \
```

Comment:
our style guides on naming macros:
https://google.github.io/styleguide/cppguide.html#Macro_Names
https://opentitan.org/book/doc/contributing/style_guides/c_cpp_coding_style.html#function-enum-struct-and-typedef-naming

applies below too